### PR TITLE
test(vmm): Add test for consistency of static CPU templates

### PR DESCRIPTION
**Pull request by zulinx86**
## Changes

Adds a test that verify consistency of static CPU templates between
hardcoded ones and JSON ones in order not to forget to update one or the other.

## Reason

...

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- ~~[ ] If a specific issue led to this PR, this PR closes the issue.~~
- [x] The description of changes is clear and encompassing.
- ~~[ ] Any required documentation changes (code and docs) are included in this PR.~~
- ~~[ ] API changes follow the [Runbook for Firecracker API changes][2].~~
- ~~[ ] User-facing changes are mentioned in `CHANGELOG.md`.~~
- [x] All added/changed functionality is tested.
- ~~[ ] New `TODO`s link to an issue.~~
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: src/vmm/tests/cpu_templates.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::PathBuf;
```
Comment (kalyazin) [inline src/vmm/tests/cpu_templates.rs]: nit: this file introduces a bit more than "remove unnecessary comments"

File: src/vmm/tests/cpu_templates.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    .collect()
+}
+
+fn verify_consistency_static_template(static_template: StaticCpuTemplate, filename: &str) {
+    let hardcoded_template = Some(CpuTemplateType::Static(static_template))
+        .get_cpu_template()
+        .unwrap();
+
+    let path = get_json_template_file_path(filename);
+    let json_template = serde_json::from_str(&std::fs::read_to_string(path))?;
```
Comment (kalyazin) [inline src/vmm/tests/cpu_templates.rs]: where is the "verify" part implemented?

File: src/vmm/tests/cpu_templates.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+#[test]
+#[cfg(target_arch = "x86_64")]
+fn verify_consistency_x86_static_templates() {
+    const TEMPLATES: [(StaticCpuTemplate, &str); 1] = [(StaticCpuTemplate::C3, "c3.json")];
+
+    for (template, filename) in TEMPLATES {
+        println!("{:#?}", get_json_template_file_path(filename));
```
Comment (kalyazin) [inline src/vmm/tests/cpu_templates.rs]: is this print essential?

File: src/vmm/src/cpu_config/x86_64/static_cpu_templates/mod.rs [hunk 4]
```diff
@@ -50,3 +50,40 @@ impl StaticCpuTemplate {
...
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use super::*;
+    use crate::cpu_config::templates::CustomCpuTemplate;
+
+    // Get a static CPU template stored as a JSON file.
+    fn get_json_template(filename: &str) -> CustomCpuTemplate {
```
Comment (kalyazin) [inline src/vmm/src/cpu_config/x86_64/static_cpu_templates/mod.rs:62]: This piece seems to be very similar between x86_64 and aarch64. Would it be too much to try to extract it?

File: src/vmm/tests/cpu_templates.rs (hunk 1 above)
Comment (zulinx86) [reply src/vmm/tests/cpu_templates.rs]: ah sorry, it seems the first commit "refactor(vmm): Remove unnecessary comments" contains some contents that I used in debug.

Comment (zulinx86) [reply src/vmm/tests/cpu_templates.rs]: removed the polluted part from the first commit.

File: src/vmm/tests/cpu_templates.rs (hunk 2 above)
Comment (zulinx86) [reply src/vmm/tests/cpu_templates.rs]: https://github.com/firecracker-microvm/firecracker/pull/3802#discussion_r1221300311

File: src/vmm/tests/cpu_templates.rs (hunk 3 above)
Comment (zulinx86) [reply src/vmm/tests/cpu_templates.rs]: https://github.com/firecracker-microvm/firecracker/pull/3802#discussion_r1221300311

File: src/vmm/src/cpu_config/x86_64/static_cpu_templates/mod.rs (hunk 4 above)
Comment (zulinx86) [reply src/vmm/src/cpu_config/x86_64/static_cpu_templates/mod.rs:62]: you're right. will try to define this in common test utilities.

Comment (zulinx86) [reply src/vmm/src/cpu_config/x86_64/static_cpu_templates/mod.rs]: done

**Review by kalyazin — APPROVED** [review]

**Review by mattschlebusch — APPROVED** [review]
LGTM.

**Pull request merged**